src/encoder/pb.rs

```diff
@@ -38,10 +38,12 @@ impl ProtobufEncoder {
 
 impl Encoder for ProtobufEncoder {
     fn encode<W: Write>(&self, metric_families: &[MetricFamily], writer: &mut W) -> Result<()> {
+        let mut buf = Vec::new();
```

Comment:
For the sake of performance, we should change `encode` to take a `Bytes` buffer rather than a `Write` object, but I'm not sure how big a breaking change that would be.

Reply:
@overvenus what do you think?

Reply:
ping @overvenus  do you have thoughts on this?

Reply:
I think we can leave it since ProtoBuf format will be removed in future (Prometheus has already dropped support for it).

Reply:
:+1: